core/webviz/_json_page_element.py

```diff
@@ -12,14 +13,16 @@
 _json_store_init.name = 'json_store_init.js'
 
 
-def default(o):
-    if isinstance(o, numpy.int64):
-        return int(o)
-    raise TypeError
+def default(data):
+    if isinstance(data, numpy.int64):
+        return int(data)
+    if isinstance(data, (datetime.datetime, datetime.date)):
```

Comment:
If we want to be helpful, we could always call str(data) to see if it can be printed. Plotly is specific about what kind of datestrings it likes, but in other contexts we could want to use isostring. I am leaning towards just giving an error message if it is not explicitly a string, and not implicitly converting.

Reply:
https://github.com/plotly/plotly.js/blob/master/test/jasmine/tests/lib_date_test.js
Plotly has improved this according to the tests here, the documentation is not updated on the other hand.

See a plot using ISO 8601 timestamp here.
https://codepen.io/arnigeir/pen/XPaLYV

As you suggested offline I'm also reverting the previous time formatting.